crypto/engine/eng_cnf.c

```diff
@@ -29,8 +29,10 @@ static STACK_OF(ENGINE) *initialized_engines = NULL;
 
 static int int_engine_init(ENGINE *e)
 {
-    if (!ENGINE_init(e))
+    if (!ENGINE_init(e)) {
+        ENGINE_finish(e);
```

Comment:
This is wrong. ENGINE_finish() should not be called if ENGINE_init fails.

Reply:
Looking again I much a funct_ref-- buried deep down, this is a false positive.